From stable-bounces@linux.kernel.org Mon Feb 12 00:54:10 2007
Date: Mon, 12 Feb 2007 00:52:20 -0800
From: Heiko Carstens <heiko.carstens@de.ibm.com>
Subject: lockdep: forward declare struct task_struct
To: torvalds@linux-foundation.org
Cc: schwidefsky@de.ibm.com, akpm@linux-foundation.org, heiko.carstens@de.ibm.com, mingo@elte.hu, stable@kernel.org
Message-ID: <200702120852.l1C8qKJr026562@shell0.pdx.osdl.net>


From: Heiko Carstens <heiko.carstens@de.ibm.com>

3117df0453828bd045c16244e6f50e5714667a8a causes this:

In file included from arch/s390/kernel/early.c:13:
include/linux/lockdep.h:300: warning:
		"struct task_struct" declared inside parameter list
include/linux/lockdep.h:300:
		warning: its scope is only this definition or
		declaration, which is probably not what you want

Acked-by: Ingo Molnar <mingo@elte.hu>
Cc: Martin Schwidefsky <schwidefsky@de.ibm.com>
Signed-off-by: Heiko Carstens <heiko.carstens@de.ibm.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/lockdep.h |    2 ++
 1 file changed, 2 insertions(+)

--- linux-2.6.20.1.orig/include/linux/lockdep.h
+++ linux-2.6.20.1/include/linux/lockdep.h
@@ -8,6 +8,8 @@
 #ifndef __LINUX_LOCKDEP_H
 #define __LINUX_LOCKDEP_H
 
+struct task_struct;
+
 #ifdef CONFIG_LOCKDEP
 
 #include <linux/linkage.h>
